app/src/main/java/com/yacgroup/yacguide/database/tourbook/TourbookExporter.kt

```diff
@@ -41,11 +49,57 @@ class TourbookExporter(
     private val _partnersKey = "partners"
     private val _notesKey = "notes"
 
+    enum class ExportFormat {
+        eJSON,
+        eJSONVEROBSE,
+        eCSV;
+    }
+
+    var exportFormat: ExportFormat = ExportFormat.eJSON
+
     @Throws(IOException::class)
     fun exportTourbook(uri: Uri) {
+        when (exportFormat) {
+            ExportFormat.eJSON, ExportFormat.eJSONVEROBSE -> _exportAsJson(uri)
+            ExportFormat.eCSV -> _exportAsCsv(uri)
+        }
+    }
+
+    /*
+     * Export tour book as CSV file
+     *
+     * See https://commons.apache.org/proper/commons-csv/apidocs/index.html
+     */
+    @Throws(IOException::class)
+    private fun _exportAsCsv(uri: Uri) {
+        val writer = StringBuffer()
+        val csvFormat = CSVFormat.Builder.create(CSVFormat.DEFAULT).apply {
+            setHeader(*TourbookEntryVerbose.keys().map { it }.toTypedArray())
+            setTrim(true)
+        }.build()
+        CSVPrinter(writer, csvFormat).apply {
+            _db.getAscends().forEach {
+                printRecord(TourbookEntryVerbose(it, _db).values())
+            }
+            flush()
+            close()
+            _writeStrToUri(uri, writer.toString())
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun _exportAsJson(uri: Uri) {
         val jsonAscends = JSONArray()
-        _db.getAscends().map { jsonAscends.put(ascend2Json(it)) }
-        _writeStrToUri(uri, jsonAscends.toString())
+        _db.getAscends().map {
```

Comment:
Please use `forEach()`. `forEach()` does not create an output list while `map()` does.